pkg/eventemitter/eventfilter.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2024 The KEDA Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package eventemitter
+
+import (
+	"fmt"
+
+	eventingv1alpha1 "github.com/kedacore/keda/v2/apis/eventing/v1alpha1"
+)
+
+// EventFilter defines the behavior for different event handlers
+type EventFilter struct {
+	IncludedEventTypes []eventingv1alpha1.CloudEventType
+
+	ExcludedEventTypes []eventingv1alpha1.CloudEventType
+}
+
+// NewEventFilter creates a new EventFilter
+func NewEventFilter(includedEventTypes []eventingv1alpha1.CloudEventType, excludedEventTypes []eventingv1alpha1.CloudEventType) *EventFilter {
+	return &EventFilter{
+		IncludedEventTypes: includedEventTypes,
+		ExcludedEventTypes: excludedEventTypes,
+	}
+}
+
+// FilterEvent returns true if the event should be handled
+func (e *EventFilter) FilterEvent(eventType eventingv1alpha1.CloudEventType) (bool, error) {
+	if len(e.IncludedEventTypes) > 0 {
+		if len(e.ExcludedEventTypes) > 0 && !e.filterExcludedEventTypes(eventType) {
+			return false, fmt.Errorf("eventtype: %s in both included and excluded types. Cannot be filtered", eventType)
+		}
```

Comment:
I don't understand this check. We haven't tested if the event is in `IncludedEventTypes` but we are raising and error because it's included in `ExcludedEventTypes`, right? 

I think that we can just accept the preference of including over excluding inside FilterEvent because we have already validated it during the admission and this is not an error itself. But if we want to check it here, I'd just execute both filters and compare if the results are in conflict, e.g:

```golang
	includeFilter := false
	excludeFilter := false
        if len(e.IncludedEventTypes) > 0 {
   		includeFilter = e.filterIncludedEventTypes(eventType)
	}

	if len(e.ExcludedEventTypes) > 0 {
		excludeFilter = e.filterExcludedEventTypes(eventType)
	}

	if includeFilter != excludeFilter {
		return false, fmt.Errorf("eventtype: %s in both included and excluded types. Cannot be filtered", eventType)
	}

	// doesn't matter if we use `includeFilter` or `excludeFilter` because they are the same
	return includeFilter, nil
```

The problem that I see here is what will happen if users configure `Include` and `Exclude` filter but not covering all the filters. It's a case that will happen when we include new event types, but I guess that it's something acceptable as the situation is an error indeed and users have to fix it.

WDYT @tomkerkhove ?